src/widgets/attachments/attachment-document.component.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import styles from './attachment-document.css';
+
+export default function AttachmentDocument(props: AttachmentDocumentProps) {
+    function handleDelete() {
+        props.onDelete(props.id);
+    }
+
+    return (
+        <div className={styles.thumbnail}>
+          <span className={styles.iconCloseBlack} onClick={handleDelete}>
+            x
+          </span>
+          <div className={styles.thumbnailDoc}>
+            <a target="_blank" href={props.src}>
+              <img src={props.src} />
+            </a>
+          </div>
+          <div className={styles.thumbnailCap}>
+            <span className={styles.thumbnailCapText}>{props.caption}</span>
+          </div>
+        </div>
+      );
+}
+
+type AttachmentDocumentProps = {
+    src: string;
+    caption?: string;
+    type: string;
+    id: number;
+    onDelete?: any;
```

Comment:
Also don't forget to type these.

Reply:
for the function, I wasn't really sure what type it was supposed to be.

Reply:
Good point. Thought from the usage it's something like `onDelete?(id: string): void;`. Note that the TypeScript code says its optional, but the prop/function is just called.

Reply:
Thanks. I can't actually remember why I made optional. I remember having an error at one point but can't remember  the error. But I've removed the optional from `onDelete` and `caption` as they no longer need it. But for the onDelete, I left the signature as `onDelete(id: number): void;`. Before this PR is merged I'll make the ID a string. The reason it's a number now is that I temporarily initialised the elements in a array so I'm using their index to access them. Currently working on add calls to the backend where the state will be set by data from the backend and I won't need to keep track of the elements with their index.